src/content_sao.h

```diff
@@ -45,8 +45,9 @@ class UnitSAO: public ServerActiveObject
 
 	inline bool isAttached() const
 	{ return getParent(); }
-	inline bool isImmortal() const
-	{ return itemgroup_get(m_armor_groups, "immortal"); }
+
+	const bool isImmortal()
```

Comment:
the const here is useless
did you mean `bool isImmortal() const`?

also why not inline anymore?

Reply:
I accidentally removed `inline`. Re-added, sorry.

I used `const bool` because the return value doesn't need to be modified. Doesn't it work that way?

And `isImmortal() const` throws the following error:

```
src/content_sao.h:50:40: error: passing ‘const UnitSAO’ as ‘this’ argument discards qualifiers [-fpermissive]
  { return itemgroup_get(getArmorGroups(), "immortal"); }
                                        ^
```

... which is why I removed it. It seems to make sense too; I just don't understand why it worked fine before.

Reply:
Fixed by making `getArmorGroups` a `const` method.

Reply:
> I used `const bool` because the return value doesn't need to be modified. Doesn't it work that way?

No. Even if it did, forbidding the caller from changing the returned value at some point afterwards would be stupid.

Reply:
I see... I'll make the return value non-const then.